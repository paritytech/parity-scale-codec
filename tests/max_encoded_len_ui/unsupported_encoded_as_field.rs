use jam_codec::{Encode, EncodeAsRef, MaxEncodedLen};

#[derive(Encode)]
struct NotMel;

impl<'a> EncodeAsRef<'a, u32> for NotMel {
	// Obviously broken but will do for this test
	type RefType = &'a u32;
}

#[derive(Encode, MaxEncodedLen)]
struct UnsupportedEncodedAsField {
	mel: u32,
	#[codec(encoded_as = "NotMel")]
	not_mel: u32,
}

fn main() {}
