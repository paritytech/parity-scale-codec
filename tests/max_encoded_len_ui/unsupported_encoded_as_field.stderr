error[E0277]: the trait bound `NotMel: MaxEncodedLen` is not satisfied
  --> tests/max_encoded_len_ui/unsupported_encoded_as_field.rs:14:23
   |
14 |     #[codec(encoded_as = "NotMel")]
   |                          ^^^^^^^^ the trait `MaxEncodedLen` is not implemented for `NotMel`
   |
   = help: the following other types implement trait `MaxEncodedLen`:
             ()
             (TupleElement0, TupleElement1)
             (TupleElement0, TupleElement1, TupleElement2)
             (TupleElement0, TupleElement1, TupleElement2, TupleElement3)
             (TupleElement0, TupleElement1, TupleElement2, TupleElement3, TupleElement4)
             (TupleElement0, TupleElement1, TupleElement2, TupleElement3, TupleElement4, TupleElement5)
             (TupleElement0, TupleElement1, TupleElement2, TupleElement3, TupleElement4, TupleElement5, TupleElement6)
             (TupleElement0, TupleElement1, TupleElement2, TupleElement3, TupleElement4, TupleElement5, TupleElement6, TupleElement7)
           and $N others
